grails-app/services/au/org/ala/biocollect/merit/ActivityService.groovy

```diff
@@ -464,4 +465,30 @@ class ActivityService {
             return result.content?.subMap('data')  ?: result
         }
     }
+
+    /**
+     * Get linked entities such as outputs, site and documents for a list of activities.
+     */
+    List<Map> addLinkedEntitiesToActivities(List<Map> activities) {
+        if (activities) {
+            List ids = activities.activityId
```

Comment:
unsure if it's possible that an Activity has no activityId but just to be safe, a safe navigation operator would be good here.

Reply:
The if check above should take care of situations where `activities` is null. Therefore, null check operator is not of much use here.